src/Illuminate/Auth/Access/Gate.php

```diff
@@ -90,10 +90,10 @@ class Gate implements GateContract
      *
      * @param  \Illuminate\Contracts\Container\Container  $container
      * @param  callable  $userResolver
-     * @param  array  $abilities
-     * @param  array  $policies
-     * @param  array  $beforeCallbacks
-     * @param  array  $afterCallbacks
+     * @param  array<string, callable>  $abilities
+     * @param  array<class-string<\Illuminate\Database\Eloquent\Model>, class-string>  $policies
+     * @param  callable[]  $beforeCallbacks
+     * @param  callable[]  $afterCallbacks
```

Comment:
These can be narrowed down as well:
```suggestion
     * @param  (callable(\Illuminate\Contracts\Auth\Authenticatable, string, array): bool|null)[]  $beforeCallbacks
     * @param  (callable(\Illuminate\Contracts\Auth\Authenticatable, string, bool|null, array): bool|null)[]  $afterCallbacks
```